src/components/molecules/ExecutionDetails/TestExecutionDetails/TestExecutionDetailsTabs.tsx

```diff
@@ -43,14 +47,18 @@ const TestExecutionDetailsTabs: React.FC = () => {
     testName,
     testSuiteName,
     startTime,
-  } = testData;
+  } = execution;
 
   const isRunning = useIsRunning(status);
 
   const decomposedVars = decomposeVariables(variables || {});
 
   const whetherToShowArtifactsTab = executorsFeaturesMap[testType]?.includes('artifacts');
 
+  useEffect(() => {
+    setTestExecutionTabsData({execution, test: entityDetails});
+  }, [id]);
```

Comment:
It should update on `entityDetails`/`execution` change too - otherwise we will not receive updates.